kubectl-volsync/cmd/migration_create.go

```diff
@@ -74,24 +123,163 @@ func doMigrationCreate(cmd *cobra.Command, args []string) error {
+	} else {
 		return err
 	}
 
+	// Check if the pvc already present
+	destPVCExists := false
+	_, err = GetDestinationPVC(ctx, migParams)
+	if err != nil {
+		if len(migParams.Dest.Capacity) == 0 {
+			return fmt.Errorf("%w, please provide the storage capacity to create destination pvc", err)
+		}
+	} else {
+		destPVCExists = true
+	}
+
+	err = CreateDestination(ctx, migParams, destPVCExists)
+	if err != nil {
+		return fmt.Errorf("%w", err)
+	}
+
+	return nil
+}
+
+func CreateDestination(ctx context.Context, migParams *MigrationParams, destPVCExists bool) error {
+	qCap, _ := resource.ParseQuantity(migParams.Dest.Capacity)
+	accessModes := []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	rsyncSpec := &volsyncv1alpha1.ReplicationDestinationRsyncSpec{
+		ReplicationDestinationVolumeOptions: volsyncv1alpha1.ReplicationDestinationVolumeOptions{
+			CopyMethod:  volsyncv1alpha1.CopyMethodNone,
+			Capacity:    &qCap,
+			AccessModes: accessModes,
+		},
+		ServiceType: &migParams.Dest.Opts.ServiceType,
+	}
+
+	if destPVCExists {
+		rsyncSpec.DestinationPVC = &migParams.Dest.PVC
+	}
+
+	rd := &volsyncv1alpha1.ReplicationDestination{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "volsync.backube/v1alpha1",
+			Kind:       "ReplicationDestination",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "migration-destination",
+			Namespace: migParams.Dest.Namespace,
+		},
+		Spec: volsyncv1alpha1.ReplicationDestinationSpec{
+			Rsync: rsyncSpec,
+		},
+	}
+	klog.V(2).Infof("Creating ReplicationDestination in namespace %s", migParams.Dest.Namespace)
+	if err := migParams.Dest.Opts.Client.Create(ctx, rd); err != nil {
+		return err
+	}
+	klog.V(0).Infof("ReplicationDestination created in namespace %s", migParams.Dest.Namespace)
+
+	return nil
+}
+
+func prepMigrationParamsStruct(cmd *cobra.Command, relation *Relationship) (*MigrationParams, error) {
 	// Insert information into the relationship & save it
 	cap, err := cmd.Flags().GetString("capacity")
 	if err != nil {
-		return err
+		return nil, err
 	}
 	relation.Set("capacity", cap)
 
+	nsName, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("namespace", nsName)
+
+	pvcName, err := cmd.Flags().GetString("pvcname")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("pvcname", pvcName)
+
+	accessMode, err := cmd.Flags().GetString("accessmodes")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("accessmodes", accessMode)
+
+	serviceType, err := cmd.Flags().GetString("servicetype")
+	if err != nil {
+		return nil, err
+	}
+	relation.Set("serviceType", serviceType)
 	if err = relation.Save(); err != nil {
-		return fmt.Errorf("unable to save relationship configuration: %w", err)
+		return nil, fmt.Errorf("unable to save relationship configuration: %w", err)
 	}
 
 	// TODO: Set up objects on the cluster
-	_, _ = newClient("")
+	clientObject, _ := newClient("")
```

Comment:
Pass the cluster context name to `newClient` (from the PVC name's XClusterName object). That way the client will point to the correct kube cluster and not just the user's default one.